_posts/2023-01-19-blog-new-svg-engine.md

```diff
@@ -0,0 +1,201 @@
+
+Over the following weeks, text-related issues had to be fixed, which were responsible for a bunch of the remaining test issues. Transformed text did not render, which turned out to be a simple <a href="https://commits.webkit.org/255801@main">mistake</a>. More tests were upstreamed, related to <a href="https://commits.webkit.org/256502@main">compositing</a> and <a href="https://commits.webkit.org/256948@main">transformations</a>. More test coverage revealed that transform changes were not handled consistently -- it took a period of investigation to <a href="https://commits.webkit.org/256787@main">land a proper fix</a>. SVG transform / SMIL \<animateMotion\> / SMIL \<animateTransform\> / CSS transform changes are now handled consistently in LBSE, leading to proper repaints, as expected.
+
+Transformation support can be considered complete and properly handled both during the painting and layout phases. Dynamic changes at runtime are correctly triggering invalidations. However, the Web-exposed **SVG DOM API** that allows querying the transformation matrices of SVG elements, such as `getCTM()` and `getScreenCTM()`, was still missing. By **November 2022** a complete implementation <a href="https://commits.webkit.org/256862@main">was upstreamed</a>, that utilized the new `SVGLayerTransformComputation` class to construct the desired transformation matrices. This way the same internal API is used for painting/layout/hit-testing and implementing the **SVG DOM accessors**.
+
+By **December 2022** LBSE was in a good shape: most important architectural changes were upstreamed and the most basic features were implemented. The year closed with a proposed <a href="https://github.com/WebKit/WebKit/pull/7482">patch</a> that will avoid re-layout when an element's transform changes. The legacy SVG engine always needs a re-layout if transform changes, as the size of each ancestor can depend on the presence of transformations on the child elements -- a bad design decision two decades ago that LBSE will resolve. Only repainting should happen, but no layouts, in LBSE.
+
+Let's move on to 2023, and recap what's still missing in LBSE.
+
+### Next steps
+
+Besides fixing all remaining test regressions (see <a href="https://github.com/WebKit/WebKit/blob/main/LayoutTests/platform/mac-ventura-wk2-lbse-text/TestExpectations">`LayoutTests/platform/mac-ventura-wk2-lbse-text/TestExpectations`</a>) "SVG resources" are missing in LBSE. That includes all "paint servers" and advanced painting operations: there is no support for linear/radial gradients, no support for patterns, and no support for clipping/masking and filters. 
```

Comment:
Here I changed “and missing support for clipping/masking and filters” to “and no support for clipping/masking and filters”.  If that’s inaccurate because there’s some limited support for clipping/masking and filters, change “no support” to “partial support” or “limited support”.

Reply:
That's correct - "no support" is the right choice.